scripts/kind-with-registry.sh

```diff
@@ -155,7 +159,13 @@ EOF
     AZURE_IDENTITY_ID_PRINCIPAL_ID=$(az identity show -n "${USER_IDENTITY}" -g "${AZWI_RESOURCE_GROUP}" --query principalId -o tsv)
 
     echo "${AZURE_IDENTITY_ID}" > "${AZURE_IDENTITY_ID_FILEPATH}"
-    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Owner" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal --output none --only-show-errors; do
+    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Owner" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal; do
+      sleep 5
+    done
+    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Storage Account Contributor" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal; do
```

Comment:
Do we know exactly where this permission is needed? Does this identity need to do more than just download things from a storage account that isn't already covered by `Owner`? And do we have any mechanism set up yet to clean up stale role assignments? This change will probably mean we start hitting the limit about 3x faster.

Reply:
Yes, *something* like this is needed to accommodate Windows conformance tests which perform Azure storage-dependant k8s functional tests.

We should be able to economize this down to a single role for that, but I wanted to get green coverage ASAP.

If we're able to prove in a follow-up PR that the Storage Blob role is not needed, great.

And yes we are going to want to figure out role assignment cleanup very soon.

cc @jsturtevant @ritikaguptams